### PR TITLE
[8.0] port users_ldap_groups to 8.0

### PR DESCRIPTION
**Pull request by giacomos**

### CODE REVIEW

File: users_ldap_groups/__init__.py [hunk 1]
```diff
@@ -19,4 +19,4 @@
 #
 ##############################################################################
 
-import users_ldap_groups
+import users_ldap_groups  # noqa
```
Comment (hbrunn) [inline users_ldap_groups/__init__.py]: do `from . import`, then you don't need the noqa-comment, which is evil

File: users_ldap_groups/__openerp__.py [hunk 2]
```diff
@@ -30,13 +30,13 @@
```
Comment (hbrunn) [inline users_ldap_groups/__openerp__.py]: move this to README.rst, using https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst as template

File: users_ldap_groups/users_ldap_groups.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import users_ldap_groups_operators
+import inspect
+
+
+class CompanyLDAPGroupMapping(models.Model):
+    _name = 'res.company.ldap.group_mapping'
+    _rec_name = 'ldap_attribute'
+    _order = 'ldap_attribute'
+
+    def _get_operators(self):
```
Comment (hbrunn) [inline users_ldap_groups/users_ldap_groups.py:63]: I've never been really happy with that, did you consider reimplementing the operators with models.AbstractModel?

File: users_ldap_groups/users_ldap_groups.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            lambda cls:
+            inspect.isclass(cls) and
+            cls != users_ldap_groups_operators.LDAPOperator)
+        for name, operator in members:
+            operators.append((name, name))
+        return tuple(operators)
+
+    ldap_id = fields.Many2one('res.company.ldap', 'LDAP server', required=True)
+    ldap_attribute = fields.Char(
+        'LDAP attribute', size=64,
```
Comment (hbrunn) [inline users_ldap_groups/users_ldap_groups.py]: `size` is obsolete

File: users_ldap_groups/users_ldap_groups.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        help='The LDAP attribute to check.\n'
+             'For active directory, use memberOf.')
+    operator = fields.Selection(
+        _get_operators, 'Operator',
+        help='The operator to check the attribute against the value\n'
+        'For active directory, use \'contains\'', required=True)
+    value = fields.Char(
+        'Value', size=1024,
```
Comment (hbrunn) [inline users_ldap_groups/users_ldap_groups.py]: `size` is obsolete

File: users_ldap_groups/users_ldap_groups.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            return user_id
+        userobj = self.env['res.users']
+        user = userobj.browse(user_id)
+        logger = logging.getLogger('users_ldap_groups')
+        if self.only_ldap_groups:
+            logger.debug('deleting all groups from user %d' % user_id)
+            user.write({'groups_id': [(5, )]})
+
+        for mapping in this.group_mappings:
+            operator = mapping.operator
```
Comment (hbrunn) [inline users_ldap_groups/users_ldap_groups.py]: this line doesn't seem to do anything

**Comment by hbrunn** [comment]
Thanks for taking the time to port this!

**Comment by giacomos** [comment]
@hbrunn thanks for your review. I will apply your feedbacks asap.

**Comment by giacomos** [comment]
@hbrunn your feedbacks have been applied.

File: users_ldap_groups/__openerp__.py [hunk 7]
```diff
@@ -27,28 +27,6 @@
     "license": "AGPL-3",
     "description": """
```
Comment (hbrunn) [inline users_ldap_groups/__openerp__.py]: replace `description` by `summary` to have the following sentence show up beneath the module's name in kanban view

File: users_ldap_groups/users_ldap_groups.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+##############################################################################
+
+from openerp import models
+from openerp import fields
+from openerp import api
+import logging
+from string import Template
+
+
+class LDAPOperator(models.TransientModel):
```
Comment (hbrunn) [inline users_ldap_groups/users_ldap_groups.py]: why not `models.AbstractModel` (`models.TransientModel` creates a database table we don't need)

**Comment by hbrunn** [comment]
@giacomos thanks for your changes! With the operators, I had in mind to have one abstract model per operator with some self-description (which fields are required/readonly, maybe some help text) and then use that information in `fields_view_get`. But what you did seems fine to me too, this also makes it simple to add new operators, and we can do the fancy stuff afterwards. Just two little nags left.

Another thing I noticed: You used your default branch (8.0) to make this PR. In the future, better create a new branch for every PR, because now if you push something to your default branch, you'll mess up this PR, which will make it impossible to merge afterwards.

**Comment by giacomos** [comment]
@hbrunn nags gone. You are also right with the suggestion of using separate branches. I'll definitely do it next time. Thanks so much for you review, I learned a lot. Cheers.

**Comment by hbrunn** [comment]
you're welcome! :+1:

**Comment by gurneyalex** [comment]
could you rebase your PR to benefit from the recent runbot fixes?

**Comment by giacomos** [comment]
@gurneyalex rebased

**Comment by giacomos** [comment]
@hbrunn @gurneyalex can this PR be merged or there's something missing?

**Comment by lmignon** [comment]
:+1:  (code review)

File: users_ldap_groups/README.rst [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    :alt: License: AGPL-3
+
+users_ldap_groups
+=================
+
+Adds user accounts to groups based on rules defined by the administrator.
+
+
+Usage
+=====
```
Comment (max3903) [inline users_ldap_groups/README.rst:11]: Should be the "Configuration" section.

Otherwise
:+1:

**Comment by hbrunn** [comment]
@giacomos until recently, there were not enough thumbs up, but now it's fine to merge

**Pull request merged**